streams/src/main/java/org/apache/kafka/streams/state/internals/MemoryNavigableLRUCache.java

```diff
@@ -66,6 +68,20 @@ public MemoryNavigableLRUCache(final String name, final int maxCacheSize) {
                 .subMap(from, true, to, true).descendingKeySet().iterator(), treeMap));
     }
 
+    @Override
+    public <PS extends Serializer<P>, P> KeyValueIterator<Bytes, byte[]> prefixScan(final P prefix, final PS prefixKeySerializer) {
```

Comment:
Can you add a test or two for this in `InMemoryLRUCacheStoreTest`? Alternatively, I think you should be able to just collect all the tests in `InMemoryKeyValueStoreTest` and `CachingInMemoryKeyValueStoreTest` and move them over to
`AbstractKeyValueStoreTest` instead. That way you get test coverage for both of those plus a handful of other store classes at once, without having to copy the same test over and over across a bunch of different files.

Reply:
1) I guess the reason I didn't add test in `InMemoryLRUCacheStoreTest` was that the key type defined here is Integer and prefixScan like range/reverseRange relies on byte lexicographical ordering and not on the ordering of the key type. So, IMO this test won't yield much for this case. If needed, I can add it.But, I think the tests for reverse and reverseRange are also missing from this class for the same reason or it could just be a miss :) 
2) Regarding the merging of test classes into one, I see that both those classes override the `createKeyValueStore` method differently, so merging them as per me would need some analysis. Is it ok if we create a separate ticket and tackle it there?

Reply:
1) Could you just treat them as Bytes all the same, and just convert to/from an Integer before putting/getting them from the store? That way you're still just handling Bytes like you are in this test, it just goes through an extra layer of de/serialization. Should be able to more or less copy over the existing tests with just a bit of extra code. Can you try this, in a followup PR?
2) Yes, I was just suggesting to merge them as a possible way to make things easier and do less work, if it's going to be more then please do file a separate ticket for it.